IdentityCore/src/cache/MSIDTokenCacheKey.m

```diff
@@ -169,4 +156,135 @@ + (NSString *)familyClientId:(NSString *)familyId
     return [NSString stringWithFormat:@"foci-%@", familyId];
 }
 
+- (BOOL)isEqualToTokenCacheKey:(MSIDTokenCacheKey *)key
+{
+    if (!key)
+    {
+        return NO;
+    }
+    
+    BOOL result = YES;
+    result &= (!self.account && !key.account) || [self.account isEqualToString:key.account];
+    result &= (!self.service && !key.service) || [self.service isEqualToString:key.service];
+    result &= (!self.type && !key.type) || [self.type isEqualToNumber:key.type];
+    
+    return result;
+}
+
+#pragma mark - NSObject
+
+- (BOOL)isEqual:(id)object
+{
+    if (self == object)
+    {
+        return YES;
+    }
+    
+    if (![object isKindOfClass:MSIDTokenCacheKey.class])
+    {
+        return NO;
+    }
+    
+    return [self isEqualToTokenCacheKey:(MSIDTokenCacheKey *)object];
+}
+
+- (NSUInteger)hash
+{
+    NSUInteger hash = 17;
```

Comment:
ADTokenCacheKey uses a different hashing method. Can we ensure it won't be a problem for backward compatibility?

Reply:
I don't see when it will be a problem. ADTokenCacheKey & MSIDTokenCacheKey are 2 different classes so we will never compare them. Also hash is not serialized/deserialized to/from disc so we should be good. Please let me know if missed something.

Reply:
I was thinking about scenario, when two macOS apps are reusing the same cache, one app is using older ADAL, another new ADAL. But you’re right, as long as we’re not reusing the same cache in memory and always deserializing/serializing it fully, it should be fine.